dpsprep/outline.py

```diff
@@ -1,23 +1,33 @@
 from loguru import logger
 from pdfrw import PdfName, PdfDict, IndirectPdfDict
 import djvu.sexpr
+import re
 
 from .sexpr import SExpressionVisitor
 
 
 # Based on
 # https://github.com/pmaupin/pdfrw/issues/52#issuecomment-271190546
 class OutlineTransformVisitor(SExpressionVisitor):
+    def __init__(self, toc_pg_offset : int, total_pages : int):
+        self.toc_pg_offset = toc_pg_offset
+        self.total_pages = total_pages
+        super().__init__()
     def visit_plain_list(self, node: djvu.sexpr.StringExpression, parent: IndirectPdfDict):
         title, page, *rest = node
         # I have experimentally determined that we need to translate page indices. -- Ianis, 2023-05-03
         try:
-            page_number = int(page.value[1:]) - 1
+            m = re.search(r'[0-9]+', page.value)
+            page_number = int(m.group(0), base=10) + self.toc_pg_offset
         except ValueError:
             # As far as I understand, python-djvulibre doesn't support Djvu's page titles. -- Ianis, 2023-12-09
             logger.warning(f'Could not determine page number from the page title {page.value}.')
             return
 
+        if page_number < 0 or page_number >= self.total_pages:
+            logger.warning(f'The page title \'{page.value}\' will be translated to an invalid number {page_number}.')
```

Comment:
"will be" implies that this will happen; and you `return` specifically for it not to happen.

Reply:
Updated.